### DISCUSSION TITLE
Changing Fan speed?

### DISCUSSION THREAD

**Discussion by jmerifjKriwe** [pail23/stiebel_eltron_isg_component#58] [General]
Hi,

is there any chance to get an option to change the fan speed? I mean switching between 0, 1, 2 and 3.
This would be really cool for some use cases which are not possible by ISG itself.

Best regards

**Comment by mintar** [comment]
I have the same question. I think it should be possible; at least the [software documentation](https://www.stiebel-eltron.de/toolbox/content/docs/anleitungen/installation/ISG_Modbus/321798-44755-9770_ISG%20Modbus_de_en_fr_it_nl_cs_sk_pl_hu.pdf) says it can be controlled via Modbus:

> The following functions are available with the Modbus software:
> - Selecting operating modes
> - Selecting set temperatures
> - **Switching fan levels**
> - Selecting set DHW temperature
> - Calling up current values and system data

**Reply by mintar** [reply]
Found it! It's supported since March 2023: https://github.com/pail23/stiebel_eltron_isg_component/commit/7afd0a203aeb8630e948f5ec0823fec8b8006f5a

**Reply by jmerifjKriwe** [reply]
But how did you create the climate entity?
Seems I'm missing this one :-(

**Reply by mintar** [reply]
I didn't, I don't even have an ISG device! I'm planning to buy one and trying to make sure that it has all the features I want. So you still haven't found a solution? Maybe somebody else can help?

**Comment by Optic00** [comment]
Closing this with the current model-specific status. Writable fan level entities are implemented for the LWZ family. If they are missing on an LWZ installation, or support is needed for another controller family, please open a focused issue with the exact model and register evidence from the ISG.
